tools/SourceKit/tools/sourcekitd/bin/InProc/sourcekitdInProc.cpp

```diff
@@ -80,9 +80,16 @@ UIdent sourcekitd::UIdentFromSKDUID(sourcekitd_uid_t uid) {
 }
 
 std::string sourcekitd::getRuntimeLibPath() {
-  // FIXME: Move to an LLVM API. Note that libclang does the same thing.
 #if defined(_WIN32)
-#error Not implemented
+  MEMORY_BASIC_INFORMATION mbi;
+  char path[MAX_PATH + 1];
+  if (!VirtualQuery(static_cast<void *>(sourcekitd_initialize), &mbi,
+                    sizeof(mbi)))
+    llvm_unreachable("call to VirtualQuery failed");
+  if (!GetModuleFileNameA(static_cast<HINSTANCE>(mbi.AllocationBase), path,
+                          MAX_PATH))
+    llvm_unreachable("call to GetModuleFileNameA failed");
+  return llvm::sys::path::parent_path(path);
```

Comment:
This piece of code is essentially duplication of logic from `clang/tools/libclang/CIndexer.cpp` which needs to do the same thing. We should have a common API to use here. I'm not going to ask you to update clang to provide the common API, but could you add a FIXME to say that we need to avoid code duplication and use a common API for this functionality ?